tests/testthat/test-derive_param_confirmed_bor.R

```diff
@@ -0,0 +1,237 @@
+
+adsl <- tibble::tribble(
+  ~USUBJID, ~TRTSDTC,
+  "1",      "2020-01-01",
+  "2",      "2019-12-12",
+  "3",      "2019-11-11",
+  "4",      "2019-12-30",
+  "5",      "2020-01-01",
+  "6",      "2020-02-02",
+  "7",      "2020-02-02",
+  "8",      "2020-04-01",
+  "9",      "2020-02-01"
+) %>%
+  mutate(
+    TRTSDT = lubridate::ymd(TRTSDTC),
+    STUDYID = "XX1234"
+  )
+
+adrs <- tibble::tribble(
+  ~USUBJID, ~ADTC,        ~AVALC,
+  "1",      "2020-01-01", "PR",
+  "1",      "2020-02-01", "CR",
+  "1",      "2020-02-16", "NE",
+  "1",      "2020-03-01", "CR",
+  "1",      "2020-04-01", "SD",
+  "2",      "2020-01-01", "SD",
+  "2",      "2020-02-01", "PR",
+  "2",      "2020-03-01", "SD",
+  "2",      "2020-03-13", "CR",
+  "3",      "2019-11-12", "CR",
+  "3",      "2019-12-02", "CR",
+  "3",      "2020-01-01", "SD",
+  "4",      "2020-01-01", "PR",
+  "4",      "2020-03-01", "SD",
+  "4",      "2020-04-01", "SD",
+  "4",      "2020-05-01", "PR",
+  "4",      "2020-05-15", "NON-CR/NON-PD",
+  "5",      "2020-01-01", "PR",
+  "5",      "2020-01-10", "SD",
+  "5",      "2020-01-20", "PR",
+  "5",      "2020-05-15", "NON-CR/NON-PD",
+  "6",      "2020-02-06", "PR",
+  "6",      "2020-02-16", "CR",
+  "6",      "2020-03-30", "PR",
+  "7",      "2020-02-06", "PR",
+  "7",      "2020-02-16", "CR",
+  "7",      "2020-04-01", "NE",
+  "9",      "2020-02-16", "PD",
+  "9",      "2020-03-06", "SD"
+) %>%
+  mutate(PARAMCD = "OVR") %>%
+  bind_rows(tibble::tribble(
+    ~USUBJID, ~ADTC,        ~AVALC,
+    "9",      "2020-02-16", "Y"
+  ) %>%
+    mutate(PARAMCD = "PD")) %>%
+  mutate(
+    ADT = lubridate::ymd(ADTC),
+    STUDYID = "XX1234"
+  ) %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    by_vars = vars(STUDYID, USUBJID),
+    new_vars = vars(TRTSDT)
+  )
+
+pd_date <- admiral::date_source(
+  dataset_name = "adrs",
+  date = ADT,
+  filter = PARAMCD == "PD"
+)
+
+# derive_param_confirmed_bor ----
+## derive_param_confirmed_bor Test 1: default confirmed BOR ----
+test_that("derive_param_confirmed_bor Test 1: default confirmed BOR", {
+  actual <-
+    derive_param_confirmed_bor(
+      adrs,
+      dataset_adsl = adsl,
+      filter_source = PARAMCD == "OVR",
+      source_pd = pd_date,
+      source_datasets = list(adrs = adrs),
+      reference_date = TRTSDT,
+      ref_start_window = 28,
+      ref_confirm = 28,
+      set_values_to = vars(
+        PARAMCD = "CBOR",
+        PARAM = "Best Confirmed Overall Response by Investigator"
+      )
+    )
+
+  expected <- bind_rows(
+    adrs,
+    tibble::tribble(
+      ~USUBJID, ~ADTC,         ~AVALC,          ~AVAL,
+      "1",      "2020-02-01",  "CR",            1,
+      "2",      "2020-02-01",  "SD",            3,
+      "3",      "2020-01-01",  "SD",            3,
+      "4",      "2020-03-01",  "SD",            3,
+      "5",      "2020-05-15",  "NON-CR/NON-PD", 4,
```

Comment:
In both cases we do not expect SD because the PR, CR,, and SD assessments are not at least 28 after treatment start date. Do you handle this differently at GSK?